features/2025-07-10-offline-install-feature-spec.md

```diff
@@ -0,0 +1,222 @@
+
+Radius must support these organizations' requirements by making it easy to install, configure, and upgrade Radius when not connected to the internet, or in *air gapped* environments.
+
+### Goals
+
+* Enable platform engineers to install and upgrade Radius while not connected to the internet
+* Enable platform engineers to control what versions of all required software components are installed
+* Modify Terraform from silently being installed in the background to explicitly installed by the platform engineer
+* Increase the transparency of what software is required to run Radius
+
+### Non-Goals (out of scope)
+
+* Enhancing the Radius installation experience in other ways
+
+## Inventory of software components
+
+Radius makes the assumption that the installation environment has full access to the internet and installs many software components on behalf of the user. Below is an inventory:
+
+| #    | Component                                                    | As-Is Distribution                                           | To-Be Online Distribution                                    | To-Be Offline Distribution                             |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------ |
+| 1a   | `rad` CLI binary                                             | [install.sh](https://raw.githubusercontent.com/radius-project/radius/main/deploy/install.sh) script | No change                                                    | Enhanced `install.sh`                                  |
+| 1b   | `rad-bicep` binary                                           | `rad bicep download`                                         | Installed by `install.sh`                                    | Installed by `install.sh`                              |
+| 1c   | `bicepconfig.json`                                           | `rad init`                                                   | Created by to `install.sh`                                   | Created by to `install.sh`                             |
+| 2    | [Radius Helm chart](https://github.com/project-radius/radius/tree/main/deploy/Chart) | `rad init`, `rad install` or `helm repo add radius`          | No change                                                    | Chart manually pulled                                  |
+| 3    | Radius container images (  `ucpd`, `controller`, `applications-rp`, `dynamic-rp`, `bicep`, `dashboard`) | Downloaded via the Radius Helm chart from GHCR               | No change                                                    | Manually imported into the user's private OCI registry |
```

Comment:
We have a new container as of 0.50: pre-upgrade. Link: https://github.com/radius-project/radius/pkgs/container/pre-upgrade.